dashboard/test/controllers/api/v1/pd/workshops_controller_test.rb

```diff
@@ -111,6 +111,81 @@ class Api::V1::Pd::WorkshopsControllerTest < ::ActionController::TestCase
     assert_equal enrollment_2.code, response.find {|workshop| workshop_2.id == workshop['id']}['enrollment_code']
   end
 
+  test 'workshops_user_enrolled_in does not return future or in progress teachercon or fit workshops' do
+    teacher = create :teacher
+    sign_in(teacher)
+
+    teachercon = create(
+      :pd_workshop,
+      :teachercon,
+      :funded,
+      organizer: @organizer,
+      facilitators: [@facilitator],
+      regional_partner: @regional_partner
+    )
+
+    fit_weekend = create(
+      :pd_workshop,
+      :funded,
+      subject: Pd::Workshop::SUBJECT_CSD_FIT,
+      course: Pd::Workshop::COURSE_CSD,
+      organizer: @organizer,
+      facilitators: [@facilitator],
+      regional_partner: @regional_partner
+    )
+
+    create(:pd_enrollment, workshop: teachercon, email: teacher.email, user_id: teacher.id)
+    create(:pd_enrollment, workshop: fit_weekend, email: teacher.email, user_id: teacher.id)
+
+    session = create(:pd_session, workshop: teachercon)
+    teachercon.sessions << session
```

Comment:
note, you can instead supply [num_sessions](https://github.com/code-dot-org/code-dot-org/blob/7820519498b9be7e1ab1a95f484fe95e15e17504/dashboard/test/factories/pd_factories.rb#L20): 1 in the `create :pd_workshop` on line 118

Reply:
ooh thanks! much cleaner.